### PR TITLE
sample: refactor events for application

### PR DESCRIPTION
**Pull request by RobertGalatNordic**
move events to functions

for now only dut is compiling

TODO:
- [x] align samples
    - [x] demo
    - [x] hello
- [x] add dfu state into the application
- [x] fix unit tests

## CI parameters

```yaml
Github_actions:
  #(branch, hash, pull/XXX/head)
  NRF_revision: main

  # Do not change after creating PR
  Create_NRF_PR: false
Jenkins:
  test-sdk-sidewalk: master
```

## Description

JIRA ticket: 

## Self review

- [ ] There is no commented code.
- [ ] There are no TODO/FIXME comments without associated issue ticket.
- [ ] Commits are properly organized.
- [ ] Change has been tested.
- [ ] Tests were updated (if applicable).

### CODE REVIEW

**Review by ktaborowski — COMMENTED** [review]
looks good so far

File: samples/sid_end_device/src/sensor_monitoring/app.c [hunk 1]
```diff
@@ -163,25 +164,58 @@ static void on_sidewalk_status_changed(const struct sid_status *status, void *co
...
+		k_timer_start(&notify_timer, K_MSEC(NOTIFY_TIMER_DURATION_MS),
+			      K_MSEC(CONFIG_SID_END_DEVICE_NOTIFY_DATA_PERIOD_MS));
+	}
+}
+
+static void app_btn_dfu_state(uint32_t unused)
+{
+	ARG_UNUSED(unused);
+	static bool go_to_dfu_state = true;
+	int e = sidewalk_event_send(app_event_dfu_mode, &go_to_dfu_state, NULL);
```
Comment (ktaborowski) [inline samples/sid_end_device/src/sensor_monitoring/app.c:201]: consider split to dfu_mode_enter and _exit, insted of passing the state variable, this should simplify code

File: samples/sid_end_device/src/hello/app.c [hunk 2]
```diff
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+#include "state_notifier/state_notifier.h"
```
Comment (ktaborowski) [inline samples/sid_end_device/src/hello/app.c:7]: change suggestion:
```
#include <state_notifier/state_notifier.h>
```

File: samples/sid_end_device/src/hello/app.c [hunk 3]
```diff
@@ -213,51 +216,105 @@ static void free_sid_hello_event_ctx(void *ctx)
...
 }
 
 static int app_buttons_init(void)
 {
-	button_set_action_short_press(DK_BTN1, app_button_handler, SID_EVENT_SEND_MSG);
-	button_set_action_long_press(DK_BTN1, app_button_handler, SID_EVENT_NORDIC_DFU);
-	button_set_action_short_press(DK_BTN2, app_button_handler, SID_EVENT_CONNECT);
-	button_set_action_long_press(DK_BTN2, app_button_handler, SID_EVENT_FACTORY_RESET);
-	button_set_action(DK_BTN3, app_button_handler, SID_EVENT_LINK_SWITCH);
+	button_set_action_short_press(DK_BTN1, app_btn_send_msg, 0);
```
Comment (ktaborowski) [inline samples/sid_end_device/src/hello/app.c:313]: minor: can add sth like `#define UNUSED_ARGS (0U)` to avoid magic numbers (just suggestion)

**Review by ktaborowski — APPROVED** [review]
Nice job, looks more robust, good idea with context cleaner :)
Just a few comments below,
Also make sure there is no regression.

Please add a migration guide (might be in a new PR).

**Comment by totyz** [comment]
Stress tests in progress: https://jenkins-ncs.nordicsemi.no/job/KRKNG/job/Sidewalk/job/test-sdk-sidewalk/job/test-branch/129/

**Pull request merged**